code/game/objects/structures/extinguisher.dm

```diff
@@ -153,8 +142,23 @@
 			stored_extinguisher = null
 	qdel(src)
 
+/obj/structure/extinguisher_cabinet/update_overlays()
+	. = ..()
+	if(stored_extinguisher)
+		if(stored_extinguisher.type == /obj/item/extinguisher/mini)
+			. += "extinguisher_mini"
+		else if(stored_extinguisher.type == /obj/item/extinguisher/advanced)
```

Comment:
Could you access a variable in the objects here, instead of hardcoding it with istype checks?

Reply:
Sure, if you think having a variable on the extinguisher is better i can do that.